### PR TITLE
[POAE7-1387] Velox String Function via UDFOutputString

### PR DESCRIPTION
**Pull request by xieqi**

### CODE REVIEW

**Comment by xieqi** [comment]
@Intel-bigdata/bdf-cider-engine

File: QueryEngine/CMakeLists.txt [hunk 1]
```diff
@@ -308,7 +308,7 @@ set(QUERY_ENGINE_LIBS
   ${Arrow_LIBRARIES}
 )
 
-list(APPEND QUERY_ENGINE_LIBS ${llvm_libs} ${ZLIB_LIBRARIES})
+list(APPEND QUERY_ENGINE_LIBS ${llvm_libs} ${ZLIB_LIBRARIES} glog)
```
Comment (winningsix) [inline QueryEngine/CMakeLists.txt:311]: Why we need glog here?

File: Cider/VeloxTypeConvertUtil.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#pragma once
+
+#include <string>
+
+#include "velox/functions/UDFOutputString.h"
+
+namespace omnisci {
+namespace cider {
+
+class CiderUDFOutputString : public facebook::velox::UDFOutputString {
```
Comment (winningsix) [inline Cider/VeloxTypeConvertUtil.h:26]: Why do we need a wrapper here?

**Review by winningsix — COMMENTED** [review]
Leave some general questions.

File: QueryEngine/CMakeLists.txt (hunk 1 above)
Comment (xieqi) [reply QueryEngine/CMakeLists.txt:311]: since velox log system use glog

File: Cider/VeloxTypeConvertUtil.h (hunk 2 above)
Comment (xieqi) [reply Cider/VeloxTypeConvertUtil.h:26]: UDFOutputString is an abstract class, you need implement the reverse function.

Comment (winningsix) [reply Cider/VeloxTypeConvertUtil.h:26]: I c. So this is similar to the concept of ```class StringProxy<FlatVector<StringView>, true /*reuseInput*/>
    : public UDFOutputString {``` in Velox.

**Comment by winningsix** [comment]
LGTM. But I am wondering whether it introduces some cycle dependencies here.

File: Cider/VeloxTypeConvertUtil.h (hunk 2 above)
Comment (xieqi) [reply Cider/VeloxTypeConvertUtil.h:26]: yes

**Comment by winningsix** [comment]
LGTM. Let's rebase and merge this.

**Review by yma11 — APPROVED** [review]
LGTM.

**Review by winningsix — APPROVED** [review]

**Pull request merged**